dist/tools/isrvecgen/isrvecgen.py

```diff
@@ -0,0 +1,222 @@
+        if (i % 0x2) == 0 and i != 0:
+            res += " || \\\n   "
+        elif i & 0x1:
+            res += " ||"
+        res += ' defined(CPU_MODEL_{})'.format(cpu.upper())
+        i += 1
+
+    return res + "\n"
+
+
+def pull_common_vectors(specific):
+    res = dict()
+
+    for i in range(0, 256):
+        tmp = None
+        match = False
+
+        for cpulist in specific:
+            if not i in cpulist["vec"]:
+                match = False
+                break
+            else:
+                if not tmp:
+                    tmp = cpulist["vec"][i]
+                    match = True
+                if cpulist["vec"][i] != tmp:
+                    match = False
+                    break
+
+        if match:
+            res[i] = copy.copy(tmp)
+            for cpulist in specific:
+                del cpulist["vec"][i]
+
+    return res
+
+def generate_weak_defaults(common, specific):
+    funcs = set()
+    for num in common:
+        funcs.add(common[num]["name"])
+    for m in specific:
+        for i in m["vec"]:
+            funcs.add(m["vec"][i]["name"])
+
+    weaks = ""
+    for f in sorted(funcs):
+        weaks += 'WEAK_DEFAULT void {}(void);\n'.format(f)
+    return weaks
+
+def fmt_table_entry(num, name, comment):
+    return '    [{:>2}] = {:<26}{}\n'.format(num, name + ",", comment)
+
+def generate_table(common, specific):
+    table = "/* CPU specific interrupt vector table */\n"
+    table += "ISR_VECTOR(1) const isr_t vector_cpu[CPU_IRQ_NUMOF] = {\n"
+    table += "    /* shared vectors for all family members */\n"
+
+    for num in sorted(common):
+        vec = common[num]
+        table += fmt_table_entry(num, vec["name"], vec["comment"])
+        last_num = num
+
+    if len(specific) > 1:
+        table += "\n"
+
+    started = False
+    for cpulist in specific:
+        prefix = "#if" if not started else "#elif"
+
+        # print("length is", len(cpulist["vec"]))
+        if len(cpulist["vec"]) > 1:
+            started = True
+            table += depstr(cpulist["cpus"], prefix)
+            for num in cpulist["vec"]:
+                vec = cpulist["vec"][num]
+                table += fmt_table_entry(num, vec["name"], vec["comment"])
+                table += ''
+
+    if len(specific) > 1:
+        table += "#endif\n"
+    table += "};"
+    return table
+
+
+if __name__ == "__main__":
+    # Define some command line args
+    p = argparse.ArgumentParser()
+    p.add_argument("cpu_conf", help="path to a cpu_conf.h file")
+    args = p.parse_args()
+
+    if not os.path.isfile(args.cpu_conf):
+        sys.exit("Error: no cpu_conf.h file found")
+
+    # get a list of available header files
+    parse_cpuconf(args.cpu_conf)
+
+    # parse each header file
+    specific = list()
+
+    for h in headermap.keys():
+        specific.append(parse_header(h, headermap[h]))
+
+    # list items by CPU name
+    specific = sorted(specific, key=lambda k: sorted(list(k["cpus"]))[0])
```

Comment:
This is personal opinion but in general I tend to avoid this kind of oneliner, for code readability reasons.

Reply:
I think this is perfectly readable and one very good example why lamdas are so useful :-)